.github/pull_request_template.md

```diff
@@ -0,0 +1,145 @@
+- [ ] Might this change have any ramifications for other parts of the system, or backward compatibility?
+
+### Security and Data Privacy
+
+- [ ] Does this code open the software up for security vulnerabilities?
+- [ ] Does this code change reveal some secret information like keys, passwords, or usernames (e.g., checked-in credentials)?
+
+### Performance
+
+- [ ] Do you think this code change will impact system performance in a negative way?
+- [ ] Do you see any potential to improve the performance of the code?
+
+### Usability
+
+- [ ] Is the proposed solution well designed from a usability perspective?
+- [ ] Is the API well documented?
+- [ ] Is the API/UI intuitive to use?
+
+### Testing
+
+- [ ] Does it have enough automated tests (unit/integration/system tests)?
+- [ ] Are there some test cases, input or edge cases that should be tested in addition?
+
+### Readability
+
+- [ ] Was the code easy to understand?
+- [ ] Can the readability of the code be improved by smaller methods?
+- [ ] Can the readability of the code be improved by different function/method or variable names?
+- [ ] Is the code located in the right file/folder/package?
+- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
+- [ ] Is the data flow understandable?
+- [ ] Are there redundant comments?
+- [ ] Could some comments convey the message better?
+- [ ] Would additional comments make the code more understandable?
+- [ ] Could some comments be removed by making the code itself more readable?
+- [ ] Is there any commented out code?
+
+### Experts Opinion
+
+- [ ] Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be committed?
+- [ ] Will this code change impact different teams?
+- [ ] Should they have a say on the change as well?
+
+## Reviewer: 
+
+### Implementation
+
+- [ ] Does this code change accomplish what it is supposed to do?
+- [ ] Can this solution be simplified?
+- [ ] Does this change add unwanted compile-time or run-time dependencies?
+- [ ] Was a framework, API, library, service used that should not be used?
+- [ ] Was a framework, API, library, service not used that could improve the solution?
+- [ ] Is the code at the right abstraction level?
+- [ ] Is the code modular enough?
+- [ ] Would you have solved the problem in a different way that is substantially better in terms of the code’s maintainability, readability, performance, security?
+- [ ] Does similar functionality already exist in the codebase? If so, why isn’t this functionality reused?
+- [ ] Are there any best practices, design patterns or language-specific patterns that could substantially improve this code?
+
+### Logic Errors and Bugs
+
+- [ ] Can you think of any use case in which the code does not behave as intended?
+- [ ] Can you think of any inputs or external events that could break the code?
+- [ ] Error Handling and Logging
```

Comment:
As above.

Reply:
Fixed in 6c75d54dd221700164ceb40b85a729e3547f8533.